tests/unit/defrag.cpp

```diff
@@ -0,0 +1,522 @@
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MODULE DEFRAG library test
+
+#include <boost/test/unit_test.hpp>
+#include <boost/filesystem.hpp>
+
+#include <future>
+
+#include "eblob_wrapper.h"
+
+#include "library/blob.h"
+#include "library/datasort.h"
+#include "library/list.h"
+
+#include "eblob/eblob.hpp"
+
+#include <vector>
+#include <utility>
+#include <random>
+#include <iostream>
+#include <functional>
+
+#include <sysexits.h>
+#include <err.h>
+
+
+void init_datasort_cfg(eblob_wrapper &wrapper, datasort_cfg &cfg) {
+	memset(&cfg, 0, sizeof(cfg));
+	cfg.b = wrapper.get();
+	cfg.log = cfg.b->cfg.log;
+}
+
+
+std::vector<char> generate_random_data(std::mt19937 &gen, std::uniform_int_distribution<unsigned> &dist) {
+	size_t datasize = 2 * (1 << 20); // 1Mib
+	if (dist(gen) % 10 != 0) {
+		datasize = 1 + dist(gen) % (1000); // <= 1KiB
```

Comment:
Why `(1000)` (not `1000`, not `(1 << 10)`)?

Reply:
you right